### DISCUSSION TITLE
ForceFeedback

### DISCUSSION THREAD

**Discussion by archiebaxter** [cyberbotics/webots#6018] [Q&A]
I have been trying for some time to get ForceFeedback on my linear arm_motor in this robotic arm example. I only get NaN in return, does anyone know why?
[Two_Finger_Gripper (2).zip](https://github.com/cyberbotics/webots/files/11052996/Two_Finger_Gripper.2.zip)

**Comment by archiebaxter** [comment]
It's the world that is called validation with controller validation too.
I am simply trying to get a force feedback result on the linear motor to enable me to validate this model against analytic calculations for my university project. My deadline is tomorrow and I am completely lost so please may someone help.

**Comment by archiebaxter** [comment]
[Two_Finger_Gripper.zip](https://github.com/cyberbotics/webots/files/11053017/Two_Finger_Gripper.zip)

**Reply by archiebaxter** [reply]
This is my current file

**Comment by ad-daniel** [comment]
As stated by the error, `enableForceFeedback()` requires 1 parameter (the sampling period), you need to give it one: `enableForceFeedback(TIME_STEP)`

**Reply by archiebaxter** [reply]
when i do that it says that it reuires one positional argument whereas i have given two. I put enableForceFeedback(16)

**Reply by archiebaxter** [reply]
Is there anyway, if you have time, could you please edit it quickly so that it works so I can see what you have done. I have literally spent hours trying to solve this so I really don't know what to do anymore

**Reply by ad-daniel** [reply]
That's the only change I did on `Two_Finger_Gripper.zip` and it works for me

**Reply by archiebaxter** [reply]
I have just tried that and it doesnt work for me
Could you zip the file you edited and send it back to me please

**Reply by archiebaxter** [reply]
Really sorry but it doesn't seem to work for me

**Reply by archiebaxter** [reply]
would it be arm_motor.enableForceFeedback(16) then?

**Reply by archiebaxter** [reply]
gives me this error:
    arm_motor.enableForceFeedback(16)
TypeError: Motor.enableForceFeedback() takes 1 positional argument but 2 were given
WARNING: 'validation' controller exited with status: 1.

**Reply by ad-daniel** [reply]
Seems to be broken in R2023a, but resolved already in the nightly build. Install the latest nightly of R2023a (called `webots-R2023a-rev1`) from [here](https://github.com/cyberbotics/webots/releases/tag/nightly_22_3_2023): 

[Two_Finger_Gripper_V2.zip](https://github.com/cyberbotics/webots/files/11053326/Two_Finger_Gripper_V2.zip)

**Reply by archiebaxter** [reply]
Okay, ill try that. Do i simply download the webots-R2023a-rev1 file and extract all or do i have to import it into webots

**Reply by archiebaxter** [reply]
I have downloaded the nightly rebuild and have extracted the zip file but it doesn't seem to have made a difference to my error that appears with positional arguments. How do you install the file properly into my webots app on my computer

**Reply by archiebaxter** [reply]
Don't worry, solved it now thank you!!!!!

**Comment by archiebaxter** [comment]
I am trying to get force feedback data on a coupled linear motor which is used to grip objects as they are lifted off the ground by a robotic arm. However, my issue is that my value for the force used by the gripper mechanism does not change regardless of the weight of the object it is picking up. Does anyone know why this is ?  For example, in the version i have attached, the value for the gripper motor force is the same whether the red box is 10, 20, 50 or 100kg. Why is this ??
[Two_Finger_Gripper.zip](https://github.com/cyberbotics/webots/files/11109631/Two_Finger_Gripper.zip)

**Comment by archiebaxter** [comment]
@ad-daniel ?

**Comment by ShuffleWire** [comment]
I'm working on force feedback as well and stumble on your issue, so I might brainstorm with you if you wich

however in the world you send, the box is never picked up by the arm, is that normal ?
